geti_sdk/demos/predict_video.py

```diff
@@ -0,0 +1,147 @@
+    if isinstance(deployment, (str, os.PathLike)):
+        deployment = Deployment.from_folder(deployment)
+    elif not isinstance(deployment, Deployment):
+        raise ValueError(f"Unable to read deployment {deployment}")
+
+    logging.info("Load inference models")
+    deployment.load_inference_models(device=device)
+
+    # Open the video capture, this prepares the video to be ready for reading
+    cap = cv2.VideoCapture(video_path)
+
+    if cap is None or not cap.isOpened():
+        raise ValueError(f"Unable to read video from {video_path}")
+
+    # Extract original video properties
+    frame_width = int(cap.get(cv2.CAP_PROP_FRAME_WIDTH))
+    frame_height = int(cap.get(cv2.CAP_PROP_FRAME_HEIGHT))
+    fps = cap.get(cv2.CAP_PROP_FPS)
+    num_frames = int(cap.get(cv2.CAP_PROP_FRAME_COUNT))
+
+    video_duration = num_frames / fps
+    logging.info(
+        f"Input video contains {num_frames:.1f} frames, "
+        f"for a total duration of {video_duration:.1f} seconds"
+    )
+
+    t_start = time.time()
+
+    predictions = list()
+    logging.info("Running video prediction... ")
+    with logging_redirect_tqdm(tqdm_class=tqdm), tqdm(
+        total=num_frames, desc="Predicting"
+    ) as progress_bar:
+        while cap.isOpened():
+            ret, frame = cap.read()
+            if ret is True:
+                rgb_frame = cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)
+                prediction = deployment.infer(rgb_frame)
+                predictions.append((rgb_frame, prediction))
+                progress_bar.update(1)
+            else:
+                break
+
+    cap.release()
+
+    if len(predictions) == num_frames:
+        t_prediction = time.time() - t_start
+        logging.info(
+            f"Prediction completed successfully in {t_prediction:.1f} seconds. "
+        )
+
+        # Determine the output video path
+        fname, ext = os.path.splitext(video_path)
+        output_video_path = os.path.abspath(fname + "_reconstructed" + ext)
+
+        # Create a video writer to be able to save the reconstructed video
+        out_video = cv2.VideoWriter(
+            filename=output_video_path,
+            fourcc=cv2.VideoWriter_fourcc(*"mp4v"),
+            fps=fps,
+            frameSize=(frame_width, frame_height),
+        )
+
+        count = 0
+        logging.info("Running video reconstruction... ")
+        with logging_redirect_tqdm(tqdm_class=tqdm), tqdm(
+            total=num_frames, desc="Reconstructing"
+        ) as progress_bar:
+            for rgb_frame, prediction in predictions:
+                output_frame = show_image_with_annotation_scene(
+                    image=rgb_frame, annotation_scene=prediction, show_results=False
+                )
+                out_video.write(output_frame)
+                count += 1
+                progress_bar.update(1)
+        out_video.release()
+
+        if shutil.which("ffmpeg") is not None and preserve_audio is True:
+            # restore sound
```

Comment:
Maybe we could try to import the ffmpeg here? It may not be installed on the system and the user might get an error even when they don't need it. So it would be something like the following? 

```python
try:
    import ffmpeg
except ImportError:
    raise ImportError(
        "ffmpeg-python is not installed. "
        "Please install it using 'pip install ffmpeg-python' \n"
        "Also, make sure you have ffmpeg installed on your system. \n"
    )
```

Reply:
Once `ffmpeg-python` is installed(we add this in the `requirements.txt`), `import ffmpeg` doesn't give an import error. Note that `ffmpeg` and `FFmpeg` are not the same. Here, `ffmpeg` is name for `ffmpeg-python`. And `ffmpeg` that `shutil.which()` is looking for means `FFmpeg`. It's confusing, but true.

Reply:
Alright, thanks for the clarification. As long as the user does not see an ffmpeg install error unexpectedly, that's fine